opentracing/src/main/java/opentracing/TraceContextSource.java

```diff
@@ -1,48 +0,0 @@
-/**
- * Copyright 2016 The OpenTracing Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
- * in compliance with the License. You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software distributed under the License
- * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
- * or implied. See the License for the specific language governing permissions and limitations under
- * the License.
- */
-package opentracing;
-
-import java.util.Map;
-
-/**
- * Long-lived interface that knows how to create a root {@linkplain TraceContext} and encode/decode
- * any other.
- */
-public interface TraceContextSource {
-
-  /**
-   * Encodes or Decodes a {@linkplain TraceContext trace context} in binary or text formats.
-   */
-  TraceContextCodec codec();
-
-  /**
-   * Create a TraceContext which has no parent (and thus begins its own trace).
-   *
-   * <p>A TraceContextSource must always return the same type in successive calls to
-   * NewRootTraceContext().
-   */
-  TraceContext newRoot();
-
-  /**
-   * Creates a child context for {@code parent}, and returns both that child's own
-   * TraceContext as well as any Tags that should be added to the child's Span.
-   */
-  ChildTraceContext newChild(TraceContext parent);
-
-  interface ChildTraceContext {
```

Comment:
From https://github.com/opentracing/opentracing-java/pull/7#issuecomment-173808490
> - and creates a child Span that has a parent_id tag (c.f. your removal of `ChildSpanContext`)


A child span is created (with the appropriate parent_id tag) using 

    Span span = Tracer.joinTrace("new-span-name", context);